suite-native/module-passphrase/src/components/PassphraseScreenHeader.tsx

```diff
@@ -47,13 +52,20 @@ export const PassphraseScreenHeader = () => {
     const [shouldShowWarningBottomSheet, setShouldShowWarningBottomSheet] = useState(false);
 
     const handleClose = () => {
-        dispatch(cancelPassphraseAndSelectStandardDeviceThunk());
-        navigation.navigate(RootStackRoutes.AppTabs, {
-            screen: AppTabsRoutes.HomeStack,
-            params: {
-                screen: HomeStackRoutes.Home,
-            },
-        });
+        if (isWalletRemovedOnClose) {
```

Comment:
This is not handled on `PassphraseEnterOnTrezorScreen`, right? 

What about refactoring `cancelPassphraseAndSelectStandardDeviceThunk()` to not do both? (I mean extract TrezorConnect.cancel() from it). 

OR I believe it should be possible to distinguish those two states from the redux state - remove the wallet on cancel only if the wallet was never discovered or something like that...

Reply:
Nice catch! Missed that completely 😮 

I originally did that as part of the thunk, but in the end all that remained was `TrezorConnect.cancel()` for this case so I moved it to the component instead. But I guess since you suggested it, it might actually be better implementation in the end. Let me know what you think and I might redo it